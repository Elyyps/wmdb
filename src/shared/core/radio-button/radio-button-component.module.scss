.radio-button {
  position: relative;
  &__input {
    position: absolute;
    visibility: hidden;
    &:checked ~ .radio-button__check {
      background: $brand-primary;
      border-color: $brand-primary;
      &::before {
        box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2);
      }
    }
  }
  &-count {
    color: $gray-light;
    margin-left: 4px;
  }
  &:not(:last-child) {
    margin-bottom: 2px;
  }
  &__label {
    font-size: 14px;
    display: flex;
    align-items: center;
    color: $global-color;
  }
  &__check {
    border-radius: 50%;
    background-color: $global-background;
    border: 1px solid $gray-lighter;
    height: 12px;
    width: 12px;
    z-index: 5;
    display: flex;
    align-items: center;
    position: relative;
    &::before {
      display: block;
      position: absolute;
      content: "";
      width: 4px;
      height: 4px;
      border-radius: 9px;
      left: 50%;
      top: 50%;
      transform: translate(-50%, -50%);
      background: $global-background;
    }
  }

  &__check-label {
    margin-left: 8px;
    cursor: pointer;
  }
}
