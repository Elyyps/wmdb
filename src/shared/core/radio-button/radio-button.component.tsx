import * as React from "react";
import styles from "./radio-button-component.module.scss";

export interface IRadioButtonComponentProps {
  count?: number;
  id: string;
  isChecked?: any;
  key?: any;
  label?: string;
  name: string;
  onChange?: any;
  value: string;
}

const RadioButtonComponent = (props: IRadioButtonComponentProps) => (
  <div className={styles["radio-button"]}>
    <label htmlFor={props.id} className={styles["radio-button__label"]}>
      <input
        checked={props.isChecked}
        type="radio"
        name={props.name}
        className={styles["radio-button__input"]}
        aria-checked="false"
        id={props.id}
        onChange={props.onChange}
        value={props.value}
      />
      <div className={styles["radio-button__check"]} />
      <div className={styles["radio-button__check-label"]}>
        <span>{props.label}</span>
        {props.count && <span className={styles["radio-button-count"]}>({props.count})</span>}
      </div>
    </label>
  </div>
);

export { RadioButtonComponent };
