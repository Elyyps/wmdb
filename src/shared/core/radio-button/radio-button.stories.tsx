import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { RadioButtonComponent } from "./radio-button.component";

storiesOf("RadioButton", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <RadioButtonComponent id={"2"} name={"name"} value={"value"} />);
