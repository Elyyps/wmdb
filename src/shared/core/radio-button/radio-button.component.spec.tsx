import React from "react";
import { shallow } from "enzyme";
import { RadioButtonComponent } from "./radio-button.component";

describe("[RadioButton]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<RadioButtonComponent id={"2"} name={"name"} value={"value"} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
