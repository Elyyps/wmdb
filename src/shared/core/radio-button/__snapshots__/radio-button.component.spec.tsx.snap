// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RadioButton] should render component without crashing 1`] = `
<div
  className="radio-button"
>
  <label
    className="radio-button__label"
    htmlFor="2"
  >
    <input
      aria-checked="false"
      className="radio-button__input"
      id="2"
      name="name"
      type="radio"
      value="value"
    />
    <div
      className="radio-button__check"
    />
    <div
      className="radio-button__check-label"
    >
      <span />
    </div>
  </label>
</div>
`;
